slsReceiverSoftware/src/FrameSynchronizerApp.cpp

```diff
@@ -0,0 +1,575 @@
+ */
+int main(int argc, char *argv[]) {
+
+    /**	- set default values */
+    int numReceivers = 1;
+    uint16_t startTCPPort = DEFAULT_TCP_RX_PORTNO;
+    bool printHeaders = false;
+
+    /**	- get number of receivers and start tcp port from command line
+     * arguments */
+    if (argc > 1) {
+        try {
+            if (argc == 3 || argc == 4) {
+                startTCPPort = sls::StringTo<uint16_t>(argv[1]);
+                if (startTCPPort == 0) {
+                    throw std::runtime_error("Invalid start tcp port");
+                }
+                numReceivers = std::stoi(argv[2]);
+                if (numReceivers > 1024) {
+                    cprintf(RED,
+                            "Did you mix up the order of the arguments?\n%s\n",
+                            getHelpMessage().c_str());
+                    return EXIT_FAILURE;
+                }
+                if (numReceivers == 0) {
+                    cprintf(RED, "Invalid number of receivers.\n%s\n",
+                            getHelpMessage().c_str());
+                    return EXIT_FAILURE;
+                }
+                if (argc == 4) {
+                    printHeaders = sls::StringTo<bool>(argv[3]);
+                    if (printHeaders) {
+                        printHeadersLevel = sls::logINFOBLUE;
+                    }
+                }
+            } else
+                throw std::runtime_error("Invalid number of arguments");
+        } catch (const std::exception &e) {
+            cprintf(RED, "Error: %s\n%s\n", e.what(), getHelpMessage().c_str());
+            return EXIT_FAILURE;
+        }
+    }
+
+    cprintf(RESET, "Number of Receivers: %d\n", numReceivers);
+    cprintf(RESET, "Start TCP Port: %hu\n", startTCPPort);
+    cprintf(RESET, "Print Callback Headers: %s\n\n",
+            (printHeaders ? "Enabled" : "Disabled"));
+
+    /** - Catch signal SIGINT to close files and call destructors properly */
+    struct sigaction sa;
+    sa.sa_flags = 0;                     // no flags
+    sa.sa_handler = sigInterruptHandler; // handler function
+    sigemptyset(&sa.sa_mask); // dont block additional signals during invocation
+                              // of handler
+    if (sigaction(SIGINT, &sa, nullptr) == -1) {
+        cprintf(RED, "Could not set handler function for SIGINT\n");
+    }
+
+    /** - Ignore SIG_PIPE, prevents global signal handler, handle locally,
+       instead of a server crashing due to client crash when writing, it just
+       gives error */
+    struct sigaction asa;
+    asa.sa_flags = 0;          // no flags
+    asa.sa_handler = SIG_IGN;  // handler function
+    sigemptyset(&asa.sa_mask); // dont block additional signals during
+                               // invocation of handler
+    if (sigaction(SIGPIPE, &asa, nullptr) == -1) {
+        cprintf(RED, "Could not set handler function for SIGPIPE\n");
+    }
+
+    Status stat{true, false, (unsigned long)numReceivers};
```

Comment:
Maybe we can use static_cast, or make it the correct type from the beginning? size_t would make sense right?

Reply:
Yes. Agreed